tools/deploy-to-svn.sh

```diff
@@ -1,110 +1,156 @@
-if [[ -n $( git status -s --porcelain ) ]]; then
-	echo "Uncommitted changes found."
-	echo "Please deal with them and try again clean."
+		The <plugin> may be either the name of a directory in projects/plugins/,
+		or a path to a plugin directorty or file.
+
+		The <tag> is the tag or branch name in the GitHub mirror repo to be
+		deployed.
+
+		Options:
+		  --non-interactive  Exit instead of prompting for questionable cases.
+		  --dir <dir>        Use the specified directory for the SVN checkout,
+		                     instead of creating a random directory in TMPDIR.
+	EOH
 	exit 1
-fi
+}
+
+BASE=$(cd $(dirname "${BASH_SOURCE[0]}")/.. && pwd)
+. "$BASE/tools/includes/plugin-functions.sh"
+. "$BASE/tools/includes/proceed_p.sh"
+
+# Process args.
+ARGS=()
+BUILD_DIR=
+INTERACTIVE=true
+while [[ $# -gt 0 ]]; do
+	arg="$1"
+	shift
+	case $arg in
+		--non-interactive)
+			INTERACTIVE=false
+			;;
+		--dir)
+			BUILD_DIR="$1"
+			shift
+			;;
+		--dir=*)
+			BUILD_DIR="${arg#--dir=}"
+			;;
+		--help)
+			usage
+			;;
+		*)
+			ARGS+=( "$arg" )
+			;;
+	esac
+done
 
-if [ "$1" != "HEAD" ]; then
+if $INTERACTIVE && [[ ! -t 0 ]]; then
+	echo "Input is not a terminal, forcing --non-interactive."
+	INTERACTIVE=false
+fi
+if [[ ${#ARGS[@]} -ne 2 ]]; then
+	usage
+fi
 
-	# Make sure we're trying to deploy something that's been tagged. Don't deploy non-tagged.
-	if [ -z $( git tag | grep "^$TARGET$" ) ]; then
-		echo "Tag $TARGET not found in git repository."
-		echo "Please try again with a valid tag."
-		exit 1
-	fi
+TAG="${ARGS[1]}"
+
+# Check plugin.
+process_plugin_arg "${ARGS[0]}"
+PLUGIN_NAME=$(jq --arg n "${ARGS[0]}" -r '.name // $n' "$PLUGIN_DIR/composer.json")
+MIRROR=$(jq -r '.extra["mirror-repo"] // ""' "$PLUGIN_DIR/composer.json")
+WPNAME=$(jq -r '.extra["wp-plugin-name"] // ""' "$PLUGIN_DIR/composer.json")
+FAIL=false
+if [[ -z "$MIRROR" ]]; then
+	FAIL=true
+	echo "Plugin $PLUGIN_NAME has no mirror repo. Cannot deploy." >&2
+fi
+if [[ -z "$WPNAME" ]]; then
+	FAIL=true
+	echo "Plugin $PLUGIN_NAME has no WordPress.org plugin name. Cannot deploy." >&2
+fi
+$FAIL && exit 1
+
+# Check build dir.
+if [[ -z "$BUILD_DIR" ]]; then
+	TMPDIR="${TMPDIR:-/tmp}"
+	BUILD_DIR=$(mktemp -d "${TMPDIR%/}/deploy-to-svn.XXXXXXXX")
+elif [[ ! -e "$BUILD_DIR" ]]; then
+	mkdir -p "$BUILD_DIR"
 else
-	read -p "You are about to deploy a change from an unstable state 'HEAD'. This should only be done to update string typos for translators. Are you sure? [y/N]" -n 1 -r
-	if [[ $REPLY != "y" && $REPLY != "Y" ]]
-	then
-		exit 1
+	if [[ ! -d "$BUILD_DIR" ]]; then
+		proceed_p "$BUILD_DIR already exists, and is not a directory." "Delete it?"
+	elif [[ $(ls -A -- "$BUILD_DIR") ]]; then
+		proceed_p "Directory $BUILD_DIR already exists, and is not empty." "Delete it?"
 	fi
+	rm -rf "$BUILD_DIR"
+	mkdir -p "$BUILD_DIR"
+fi
+cd "$BUILD_DIR"
+DIR=$(pwd)
+echo "Using build dir $DIR"
+
+echo "Checking mirror repo"
+git init -q -b tmp .
```

Comment:
```
Checking mirror repo
error: unknown switch `b'
usage: git init [-q | --quiet] [--bare] [--template=<template-directory>] [--shared[=<permissions>]] [<directory>]
```

Looks like git 2.24.3 is included on macOS, so we'll need to install an updated version since setting the initial branch was added until 2.28 per https://github.com/git/git/commit/8747ebb7cde9e90d20794c06e6806f75cd540142#diff-2d11ba71cea98bc69c78fee203d54eef7012c7e6c9bf0a81d0eba5d7b786b407 .

We can do something like the bash version check, but use `git --version` and if the version there is sub-2.28, then suggest upgrading and/or using `brew install git` to get an updated version for macos folks.